[Term Entry] C++  Key_comp() (#6897)
Codecademy/docs: content/cpp/concepts/maps/terms/key-comp/key-comp.md

---
Title: 'Key_Comp'
Description: 'The key_comp() method in C++ is used to return a copy of the comparison object(key_compare) that was provided when the container was constructed.'
Subjects:
  - 'Computer Science'
  - 'iOS'
  - 'Developer Tools'
  - 'Information Technology'
  - 'Code Foundations'
Tags:
  - 'Data Structures'
  - 'Data Distribution'
  - 'Data'
  - 'iOS'
CatalogContent:
  - 'path/computer-science'
---

> The **key_comp()** is used to return a copy of the comparison function object(key_compare) used by the container.  

Reviewer: ```suggestion
The **`.key_comp()`** function returns a copy of the comparison function object (typically `key_compare`) used by the container to order its keys.
```

## Syntax

```C++

> Key_compare key_comp() const;```

Author: I used myMap.key_comp()  instead of map_name.key_comp() , is that fine

- 'Key_compare' is the comparison function object

> - 'Key_comp()' is used to return a copy of the comparison  function object

Reviewer: here, please state the parameters and return value of the term. If there are no parameters, add a statement that says this function takes no parameters

## Example

> 

Reviewer: please give a one line description of the example and also wrap the example in a ```cpp block

Author: I dont understand, do you mean same as the codebyte example

#include <iostream>
#include <map>

int main() {

    std::map<int, std::string> myMap = {
        {1, "Apple"},
        {2, "Banana"},
        {3, "Orange"},
        {4, "Date"},
        {5, "Mango"}
    };

    auto comp = myMap.key_comp();

    int keyToCompare = 3;

    for (const auto& pair : myMap) {
        if (comp(pair.first, keyToCompare)) {
            std::cout << pair.first << " is less than " << keyToCompare << std::endl;
        } else if (comp(keyToCompare, pair.first)) {
            std::cout << pair.first << " is greater than " << keyToCompare << std::endl;
        } else {
            std::cout << pair.first << " is equal to " << keyToCompare << std::endl;
        }
    }

    return 0;

> }

Reviewer: indentation should be two spaces

> }

Author: Thanks. Working on it.

> 
> 
> 
> ## Output
> 

Reviewer: ```suggestion
```

Reviewer: extra lines can be removed

> The code above results in the following output

Reviewer: ```suggestion
This code results in the following output:
```

1 is less than 3
2 is less than 3
3 is equal to 3
4 is greater than 3

> 5 is greater than 3

Reviewer: wrap the output in a ```shell``` block

## Codebyte Example

> 

Reviewer: give a one line description of the code

> ```codebyte/C++

Reviewer: ```suggestion
```codebyte/cpp
```

#include <iostream>
#include <map>

int main() {

    std::map<int, std::string> myMap = {
        {1, "Apple"},
        {2, "Banana"},
        {3, "Orange"},
        {4, "Date"},
        {5, "Mango"}
    };

    auto comp = myMap.key_comp();

    int keyToCompare = 3;

    for (const auto& pair : myMap) {
        if (comp(pair.first, keyToCompare)) {
            std::cout << pair.first << " is less than " << keyToCompare << std::endl;
        } else if (comp(keyToCompare, pair.first)) {
            std::cout << pair.first << " is greater than " << keyToCompare << std::endl;
        } else {
            std::cout << pair.first << " is equal to " << keyToCompare << std::endl;
        }
    }

    return 0;

> }

Reviewer: can this example be changed, it is exactly similat to the previous one
